scripts/tests/test_zkai_attention_kv_fused_softmax_table_microprofile_gate.py

```
@@ -0,0 +1,205 @@
+        try:
+            gate.EXPECTED_MUTATION_COUNT = original_count - 1
+            with self.assertRaisesRegex(gate.FusedSoftmaxTableMicroprofileGateError, "mutation spec count drift"):
+                gate.validate_mutation_spec()
+        finally:
+            gate.EXPECTED_MUTATION_COUNT = original_count
+
+    def test_write_json_validates_before_writing(self):
+        payload = self.strip_mutation_summary(self.payload)
+        payload["aggregate"]["total_trace_rows"] = 1
+        with tempfile.TemporaryDirectory() as tmp:
+            with self.assertRaisesRegex(gate.FusedSoftmaxTableMicroprofileGateError, "trace-row total drift"):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)